### PR TITLE
Add description prop to wizard for showing tooltip on steps

### PR DESCRIPTION
**Pull request by Juliakas**
Thank you for your contribution to the iTwinUi-react project! Please make sure to complete all of the items below before submitting the PR.

## Checklist

- [x] Add meaningful tests for your component (verify that all lines are covered)
- [x] Verify that all existing tests pass
- [x] Add component features demo in Storybook (different stories)
- [x] Add an entry to the changelog for any new components and changes
- [x] Add screenshots of the key elements of the component below
- [x] Add any additional comments below describing the features you've implemented

### Tooltip
image
### Tooltip hover area
image
### Prioritizes top placement if space available (auto setting)
image

## Additional Comments
Also works for long and workflow types - hover area is within the shape as there is no outer label.

~~This PR includes a fix to storybook issue: https://itwin.github.io/iTwinUI-react/?path=/story/core-wizard--long and https://itwin.github.io/iTwinUI-react/?path=/story/core-wizard--localized-long~~

### CODE REVIEW

File: src/core/Wizard/Step.tsx [hunk 1]
```diff
@@ -31,6 +32,11 @@ export type StepProps = {
    *  Click handler on completed step.
    */
   onClick?: (clickedIndex: number) => void;
+
```
Comment (bentleyvk) [inline src/core/Wizard/Step.tsx:35]: Remove empty line.

File: src/core/Wizard/Step.tsx [hunk 2]
```diff
@@ -40,6 +46,7 @@ export const Step = ({
   totalSteps,
   type,
   onClick,
+  description: tooltip,
```
Comment (bentleyvk) [inline src/core/Wizard/Step.tsx:49]: I think `description` is ok.

File: src/core/Wizard/Wizard.tsx [hunk 3]
```diff
@@ -15,7 +15,14 @@ export type WizardLocalization = {
 export type WizardType = 'default' | 'long' | 'workflow';
 
 export type StepProperties = {
+  /**
+   * The title/label of the step.
+   */
   name: string;
+  /**
+   * A tooltip giving detailed description to this step
```
Comment (bentleyvk) [inline src/core/Wizard/Wizard.tsx:23]: Missing dot at the end.

File: src/core/Wizard/Step.tsx (hunk 1 above)
Comment (Juliakas) [reply src/core/Wizard/Step.tsx]: Removed in latest commit

File: src/core/Wizard/Step.tsx (hunk 2 above)
Comment (Juliakas) [reply src/core/Wizard/Step.tsx]: Yeah, I don't know why I used different name from prop here. I went with description in latest commit.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,13 @@
 # Changelog
 
+## [X.X.X]
+
+`2021-XX-XX`
+
+### What's new
+
+- **Added optional tooltip for `Wizard` steps.**
```
Comment (bentleyvk) [inline CHANGELOG.md:9]: Maybe mention that is `description` property in `StepProperties`.

File: src/core/Wizard/Wizard.test.tsx [hunk 5]
```diff
@@ -2,8 +2,9 @@
  * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
  * See LICENSE.md in the project root for license terms and full copyright notice.
  *--------------------------------------------------------------------------------------------*/
-import { render } from '@testing-library/react';
+import { fireEvent, render } from '@testing-library/react';
 import React from 'react';
+import { act } from 'react-dom/test-utils';
```
Comment (bentleyvk) [inline src/core/Wizard/Wizard.test.tsx:7]: Use `act` from `@testing-library/react`.

File: src/core/Wizard/Wizard.test.tsx [hunk 6]
```diff
@@ -2,8 +2,9 @@
  * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
  * See LICENSE.md in the project root for license terms and full copyright notice.
  *--------------------------------------------------------------------------------------------*/
-import { render } from '@testing-library/react';
+import { fireEvent, render } from '@testing-library/react';
 import React from 'react';
+import { act } from '@testing-library/react';
```
Comment (bentleyvk) [inline src/core/Wizard/Wizard.test.tsx:7]: I wanted you to add `act` to line 5 imports 😀

Comment (Juliakas) [reply src/core/Wizard/Wizard.test.tsx]: Just noticed that import... I have merged them now

File: src/core/Wizard/Wizard.test.tsx [hunk 7]
```diff
@@ -208,4 +208,34 @@ describe('<Wizard />', () => {
...
+            description: 'Step three tooltip',
+          },
+        ]}
+      />
+    );
+
+    const { getByText, queryByText } = render(wizard);
+
+    expect(queryByText('Step one tooltip')).toBeNull();
+    await act(async () => {
```
Comment (bentleyvk) [inline src/core/Wizard/Wizard.test.tsx:236]: Seems like these async/await is not needed. Also in the beginning of the test.

File: src/core/Wizard/Wizard.test.tsx (hunk 5 above)
Comment (Juliakas) [reply src/core/Wizard/Wizard.test.tsx]: Done

File: CHANGELOG.md (hunk 4 above)
Comment (Juliakas) [reply CHANGELOG.md]: Adjusted changelog.

File: src/core/Wizard/Wizard.tsx (hunk 3 above)
Comment (Juliakas) [reply src/core/Wizard/Wizard.tsx]: Added dot

**Review by bentleyvk — DISMISSED** [review]

File: src/core/Wizard/Wizard.test.tsx [hunk 8]
```diff
@@ -208,4 +208,34 @@ describe('<Wizard />', () => {
...
+      />
+    );
+
+    const { getByText, queryByText } = render(wizard);
+
+    expect(queryByText('Step one tooltip')).toBeNull();
+    act(() => {
+      fireEvent.mouseEnter(getByText('Step One'), { bubbles: true });
+    });
+    getByText('Step one tooltip');
```
Comment (veekeys) [inline src/core/Wizard/Wizard.test.tsx:239]: Maybe add check, if there is no description, no tooltip appears.

File: src/core/Wizard/Wizard.tsx [hunk 9]
```diff
@@ -96,6 +103,7 @@ export const Wizard = ({
               totalSteps={steps.length}
               type={type}
               onClick={onStepClick}
+              description={s.description}
```
Comment (mayank99) [inline src/core/Wizard/Wizard.tsx:106]: I'm wondering if we should use `name` when `description` is not available. Will be useful when name is not visible ("Long")

Comment (mayank99) [reply src/core/Wizard/Wizard.tsx:106]: @veekeys @bentleyvk Thoughts?

Comment (veekeys) [reply src/core/Wizard/Wizard.tsx:106]: And what would be the API if user does not want to have a tooltip?

Comment (mayank99) [reply src/core/Wizard/Wizard.tsx:106]: I guess user could pass `''` idk.

Maybe no need

File: src/core/Wizard/Step.tsx [hunk 10]
```diff
@@ -40,6 +45,7 @@ export const Step = ({
   totalSteps,
   type,
   onClick,
+  description,
```
Comment (mayank99) [inline src/core/Wizard/Step.tsx:48]: Merge conflicts

**Review by veekeys — DISMISSED** [review]
Good job!
Congrats with the first (hopefully not last) contribution in Github for iTwinUI!

**Review by mayank99 — APPROVED** [review]

**Pull request merged**